Hardware/HardwareTests/MockStripePaymentIntent.swift

```diff
@@ -0,0 +1,28 @@
+@testable import Hardware
+import StripeTerminal
+// This structs emulates the properties of SCPPaymentIntent
+// We can not mock SCPPaymentIntent directly, because its initializers
+// are annotated as NS_UNAVAILABLE
+struct MockStripePaymentIntent {
+    let stripeId: String
+    let created: Date
+    let status: StripeTerminal.PaymentIntentStatus
+    let amount: UInt
+    let currency: String
+    let metadata: [AnyHashable: Any]?
+    let charges: [SCPCharge]
+}
+
+extension MockStripePaymentIntent: StripePaymentIntent {}
+
+extension MockStripePaymentIntent {
+    static func mock() -> Self {
+        MockStripePaymentIntent(stripeId: "id",
+                                created: Date(),
+                                status: .succeeded,
+                                amount: 100,
+                                currency: "USD",
```

Comment:
Should this be `usd` ?

Reply:
Not really. It does not really matter, because It's a mock, but if it is uppercase it also helps test that the property wrapper on the currency actually returns that as lowercase.